core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages.component.ts

```diff
@@ -0,0 +1,31 @@
+import { Component } from '@angular/core';
+import { Observable } from 'rxjs/internal/Observable';
+import { DotPagesState, DotPageStore } from './dot-pages-store/dot-pages.store';
+
+@Component({
+    selector: 'dot-pages',
+    templateUrl: './dot-pages.component.html',
+    styleUrls: ['./dot-pages.component.scss'],
+    providers: [DotPageStore]
+})
+export class DotPagesComponent {
+    vm$: Observable<DotPagesState> = this.store.vm$;
+
+    constructor(private store: DotPageStore) {
+        this.store.setInitialStateData();
```

Comment:
You can lazy load the state in the store itself: https://ngrx.io/guide/component-store/initialization#lazy-initialization

Which reminds me... Do we have a loading state? design?

Reply:
- I don't think it needs to be lazy initialized

- no there is not a design for LOADING state, so if needed due to missing design up to this moment and the need to close task this asap I'll prefer to create a new task to add it later